main/inc/lib/course.lib.php

```diff
@@ -3906,17 +3906,152 @@ public static function display_courses_in_category($user_category_id, $load_dirs
+            }
+
+            if ($course_info['status'] == COURSEMANAGER) {
+                //echo Display::return_icon('teachers.gif', get_lang('Status').': '.get_lang('Teacher'), array('style'=>'width: 11px; height: 11px;'));
+            }
+        } else {
+            if ($course_info['visibility'] != COURSE_VISIBILITY_CLOSED) {
+                if ($loadDirs) {
+                    $params['right_actions'] .= '<a id="document_preview_'.$course_info['real_id'].'_0" class="document_preview" href="javascript:void(0);">'.Display::return_icon('folder.png', get_lang('Documents'), array('align' => 'absmiddle'),ICON_SIZE_SMALL).'</a>';
+                    $params['right_actions'] .= Display::div('', array('id' => 'document_result_'.$course_info['real_id'].'_0', 'class'=>'document_preview_container'));
+                } else {
+                    if ($course_info['status'] == COURSEMANAGER) {
+                        $params['right_actions'].= '<a href="'.api_get_path(WEB_CODE_PATH).'course_info/infocours.php?cidReq='.$course['code'].'">'.Display::return_icon('edit.png', get_lang('Edit'), array('align' => 'absmiddle'),ICON_SIZE_SMALL).'</a>';
+                    }
+                }
+            }
+        }
+
+        $course_title_url = '';
+        if ($course_info['visibility'] != COURSE_VISIBILITY_CLOSED || $course['status'] == COURSEMANAGER) {
+            $course_title_url = api_get_path(WEB_COURSE_PATH).$course_info['path'].'/?id_session=0';
+            $course_title = Display::url($course_info['title'], $course_title_url);
+        } else {
+            $course_title = $course_info['title']." ".Display::tag('span',get_lang('CourseClosed'), array('class'=>'item_closed'));
+        }
+
+        // Start displaying the course block itself
+        if (api_get_setting('display_coursecode_in_courselist') == 'true') {
+            $course_title .= ' ('.$course_info['visual_code'].') ';
+        }
+        $teachers = '';
+        if (api_get_setting('display_teacher_in_courselist') == 'true') {
+            $teachers = CourseManager::get_teacher_list_from_course_code_to_string($course['code'], self::USER_SEPARATOR, true);
+        }
+        $params['link'] = $course_title_url;
+        $params['icon'] = $status_icon;
+        $params['title'] = $course_title;
+        $params['teachers'] = $teachers;
+        if ($course_info['visibility'] != COURSE_VISIBILITY_CLOSED) {
+            $params['notifications'] = $show_notification;
+        }
+
+        return $params;
+    }
+
+
     /**
      * Retrieves the user defined course categories
-     * @author Patrick Cool <patrick.cool@UGent.be>, Ghent University
+     * @param string $userId
      * @return array containing all the titles of the user defined courses with the id as key of the array
      */
-    public static function get_user_course_categories()
+    public static function get_user_course_categories($userId = "")
     {
         global $_user;
```

Comment:
$_user was use in this function before I've modified it, I correct it and send you a new PR, sorry for the time you spend on it :sa: